ui/packages/ui/src/Stores/viewerSlice.ts

```diff
@@ -27,7 +29,8 @@ export const viewerSlice = createSlice({
       state.error = null;
       return state;
     },
-    setError: (state, action: PayloadAction<{ error: string }>) => {
+    setError: (state, action: PayloadAction<{ recoveryConfig: string | null; error: string }>) => {
+      state.recoveryConfig = action.payload.recoveryConfig;
```

Comment:
In `start`, should set `recoveryConfig` to `null` to prevent the chance of a prior config bleeding over to a new run (case of error happening but for some reason `recoveryConfig` is not updated)